packages/roosterjs-editor-core/lib/corePlugins/CopyPlugin.ts

```diff
@@ -0,0 +1,51 @@
+import { Editor, EditorPlugin } from 'roosterjs-editor-core';
+
+/**
+ * Copy plugin, hijacks copy events to normalize the content to the clipboard.
+ */
+export default class CopyPlugin implements EditorPlugin {
+    private editor: Editor;
+    private copyDisposer: () => void;
+
+    /**
+     * Get a friendly name of  this plugin
+     */
+    getName() {
+        return 'Copy';
+    }
+
+    /**
+     * Initialize this plugin. This should only be called from Editor
+     * @param editor Editor instance
+     */
+    public initialize(editor: Editor) {
+        this.editor = editor;
+        this.copyDisposer = editor.addDomEventHandler('copy', this.onCopy);
+    }
+
+    /**
+     * Dispose this plugin
+     */
+    public dispose() {
+        this.copyDisposer();
+        this.copyDisposer = null;
+        this.editor = null;
+    }
+
+    private onCopy = (event: Event) => {
+        // if it's dark mode...
+        if (this.editor && this.editor.isDarkMode()) {
+            // get whatever the current selection range is
+            const copyFragment = this.editor.getSelection().getRangeAt(0).cloneContents();
+
+            // revert just this selected range to light mode colors
+            const normalizerDiv = document.createElement('div');
+            normalizerDiv.appendChild(copyFragment);
+            const normalizedFragment = this.editor.getColorNormalizedContent(normalizerDiv.innerHTML);
+
+            // put it on the clipboard
+            (event as ClipboardEvent).clipboardData.setData('text/html', normalizedFragment);
+            event.preventDefault();
```

Comment:
will this prevent setting plain text clipboard data and/or image?

Reply:
You're right, with prevent default the only thing on there would be the HTML fragment. We need to think more about how to support images and other clipboard data. You can't just change one of the data lines.

Reply:
I did some inspecting and research which confirmed my suspicions:

1) Right click -> Copy image. Doesn't fire an event, at least in my limited testing. We don't hijack it here.
2) Select image, copy -> Will by hijacked, but doesn't matter since it's just a document fragment.
3) Select multiple elements, copy -> Will be hijacked, but doesn't matter since it's a document fragment.

I'll add something that sets plain text.